### PR TITLE
fix(config): remove daemon.json in clean states

### PR DESCRIPTION
**Pull request by danny-smit**
Remove the daemon.json when the clean states are called
Also extends tests to verify daemon.json content en cleanup.

### PR progress checklist (to be filled in by reviewers)

- [ ] Changes to documentation are appropriate (or tick if not required)
- [ ] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Related issues and/or pull requests

#282 
#283

### Describe the changes you're proposing

Follow up on the previous pull requests #282 and #283 to:
- Remove the daemon.json when the clean states are called
- Extend tests to verify daemon.json content en cleanup.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

File: test/integration/package/controls/config.rb [hunk 1]
```diff
@@ -11,4 +11,11 @@
     its('content') { should include 'DOCKER_OPTS="-s btrfs --dns 8.8.8.8"' }
     its('content') { should include 'export http_proxy="http://172.17.42.1:3128"' }
   end
+  describe file('/etc/docker/daemon.json') do
+    it { should be_file }
+    its('owner') { should eq 'root' }
+    its('group') { should eq 'root' }
+    its('mode')  { should cmp '0640' }
+    its('content') { should include '"insecure-registries": []'}
```
Comment (myii) [inline test/integration/package/controls/config.rb:19]: We've got a [`rubocop` violation](https://gitlab.com/saltstack-formulas/docker-formula/-/jobs/1205924726#L117) that needs to be fixed here:

```suggestion
    its('content') { should include '"insecure-registries": []' }
```

**Review by myii — CHANGES_REQUESTED** [review]
Thanks for the PR, @danny-smit.  When fixing the `rubocop` violation, would you mind rebasing on the latest commit here?  I was just in the process of adding the [latest pre-salted images](https://github.com/saltstack-formulas/docker-formula/commit/2c7fe1ad322667664eb38782bd95a311c3c387cf) when you submitted this PR!

File: test/integration/package/controls/config.rb (hunk 1 above)
Comment (danny-smit) [reply test/integration/package/controls/config.rb]: Thanks. Rebase is done and applied this at the same time.

**Review by myii — APPROVED** [review]

**Comment by myii** [comment]
Nice work, thanks @danny-smit -- merged.

**Pull request merged**